collect_app/src/main/java/org/odk/collect/android/utilities/NetworkStateProvider.java

```diff
@@ -0,0 +1,23 @@
+package org.odk.collect.android.utilities;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import org.odk.collect.android.application.Collect;
+
+public class NetworkStateProvider {
```

Comment:
This would be a nice opportunity for pulling out an `interface` I think. We could have an `interface NetworkStateProvider` and a `class ConnectivityProvider`. Means in tests we can fake without extending from a real implementation.

Reply:
Done.